MetadataExtractor/Formats/WebP/WebpRiffHandler.cs

```diff
@@ -165,17 +190,23 @@ public void ProcessChunk(string fourCc, byte[] payload)
                             reader.GetByte(4) != 0x01 ||
                             reader.GetByte(5) != 0x2A)
                             break;
-                        var width = reader.GetUInt16(6);
-                        var height = reader.GetUInt16(8);
-                        var directory = new WebPDirectory();
-                        directory.Set(WebPDirectory.TagImageWidth, width);
-                        directory.Set(WebPDirectory.TagImageHeight, height);
-                        _directories.Add(directory);
+                        width = reader.GetUInt16(6);
+                        height = reader.GetUInt16(8);
                     }
                     catch (IOException e)
                     {
-                        Debug.WriteLine(e);
+                        error = "Exception reading WebpRiff chunk 'VP8' : " + e.Message;
+                    }
+
+                    var directory = new WebPDirectory();
+                    if (error == null)
+                    {
+                        directory.Set(WebPDirectory.TagImageWidth, width);
+                        directory.Set(WebPDirectory.TagImageHeight, height);
                     }
+                    else
+                        directory.AddError(error);
+                    _directories.Add(directory);
                     break;
                 }
             }
```

Comment:
For all of the cases in WebRiffHandler, the replacement goal was to always return a directory. If data isn't valid, those Sets are avoided but AddError will ensure the problem is propagated for possible review.